extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/data/input/kafkainput/KafkaStringHeaderReader.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.data.input.kafkainput;
+
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.Headers;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class KafkaStringHeaderReader implements KafkaHeaderReader
+{
+  private static final Logger log = new Logger(KafkaStringHeaderReader.class);
+  private final Headers headers;
+  private final String headerLabelPrefix;
+  private final String encoding;
+
+  public KafkaStringHeaderReader(Headers headers,
+                                 String encoding,
+                                 String headerLabelPrefix)
+  {
+    this.headers = headers;
+    this.encoding = encoding;
+    this.headerLabelPrefix = headerLabelPrefix;
+  }
+
+  @Override
+  public Map<String, Object> read() throws IOException
+  {
+    Map<String, Object> events = new HashMap<>();
+    for (Header hdr : headers) {
+      String s = new String(hdr.value(), this.encoding);
```

Comment:
we should decide if we want to handle invalid byte sequences here or bubble the exception up. Maybe take a look at other input formats to see what the behavior is elsewhere so we are consistent here.